rosbag2_storage_default_plugins/src/rosbag2_storage_default_plugins/sqlite/sqlite_storage.cpp

```diff
@@ -82,7 +87,29 @@ void SqliteStorage::open(
     initialize();
   }
 
-  ROSBAG2_STORAGE_DEFAULT_PLUGINS_LOG_INFO_STREAM("Opened database '" << uri << "'.");
+  uri_ = uri;
+  io_flag_ = io_flag;
+  max_bagfile_size_ = max_bagfile_size;
+
+  ROSBAG2_STORAGE_DEFAULT_PLUGINS_LOG_INFO_STREAM("Opened database '" << database_name_ << "'.");
+}
+
+uint64_t SqliteStorage::get_db_size()
+{
+  struct stat stat_res;
```

Comment:
This will not compile on Windows. See: https://stackoverflow.com/questions/8991192/check-the-file-size-without-opening-file-in-c

The ifdef block should be short and localized. The code should be added to `rosbag2_storage::FilesystemHelper` following the existing style.

```c++
static uint64_t get_file_size(const std::string &file_path);

#ifdef WIN32
uint64_t get_file_size(const std::string &file_path) {
 // Windows implementation
}
#else
uint64_t get_file_size(const std::string &file_path) {
  if (stat(file_path.c_str(), &stat_res) == 0) {
    return stat_res.st_size;
  } else {
   std::stringstream ss;
   ss << "Failed to get size of file '" << file << "':" << std::strerror(errno));
   throw std::runtime_error(ss.str());
  }
}
#endif
```